Add rough tumor classification into the result table-Wilms tumor annotation (SCPCP000014)

JingxuanChen7 opened the pull request:
### Purpose/implementation Section


#### Please link to the GitHub issue that this pull request addresses.

Analysis proposal: https://github.com/AlexsLemonade/OpenScPCA-analysis/discussions/628
Issue: https://github.com/AlexsLemonade/OpenScPCA-analysis/issues/782
Follow-up of PR https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/836

#### What is the goal of this pull request?

This PR simply adds a rough tumor cell classification to the summary result table.

#### Briefly describe the general approach you took to achieve this goal.

Rough tumor cell classification is inferred based on anchor transfer results:
  - Cells annotated as "Cap mesenchyme" were labeled as tumor cells: CM should have gone since born [(Ref)](https://pmc.ncbi.nlm.nih.gov/articles/PMC4675508/)
  - Cells annotated as "Immune" or "Endothelium" were labeled as normal cells: Wilms tumor shows a tri-phasic type, meaning blastemal, stromal, and epithelial cells can be tumor cells [Ref1](https://pmc.ncbi.nlm.nih.gov/articles/PMC6076422/) [Ref2](https://www.frontiersin.org/journals/oncology/articles/10.3389/fonc.2023.1137346/full). In contrast, "Immune" or "Endothelium" cells are regarded as normal cells.
  - Other cells were labeled as "unknown"

#### If known, do you anticipate filing additional pull requests to complete this analysis module?

No.

### Results


#### What is the name of your results bucket on S3?

` s3://researcher-009160072044-us-east-2/cell-type-wilms-tumor-14/results/01_anchor_transfer_seurat/summary_results.csv`


#### What types of results does your code produce (e.g., table, figure)?
Table in `csv`.

#### What is your summary of the results?




### Provide directions for reviewers


#### What are the software and computational requirements needed to be able to run the code in this PR?

N.A.



#### Are there particularly areas you'd like reviewers to have a close look at?

N.A.

#### Is there anything that you want to discuss further?

N.A.

### Author checklists

_Check all those that apply._
_Note that you may find it easier to check off these items after the pull request is actually filed._


#### Analysis module and review

- [x] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [x] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [x] The analytical code is documented and contains comments.
- [x] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [x] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [ ] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [x] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

jashapiro commented on analyses/cell-type-wilms-tumor-14/scripts/summary_results.R:
```diff
@@ -89,7 +89,18 @@ arg_df <- tidyr::expand_grid(
-  purrr::reduce(left_join)
+  purrr::reduce(left_join) %>%
+  mutate(tumor_cell_classification_RNA = case_when(
+    cell_type_assignment_RNA_compartment == "immune" | cell_type_assignment_RNA_compartment == "endothelium" ~ "normal",
+    cell_type_assignment_RNA_compartment == "fetal_nephron" & cell_type_assignment_RNA_celltype == "Cap mesenchyme" ~ "tumor",
+    TRUE ~ "unknown"
```
For a bit more clarity (and possibly efficiency), I would use `%in%` here, and separately define the classes that you are using. 

I also updated to using the new `case_when()` `.default` argument.

```suggestion
    cell_type_assignment_RNA_compartment %in% normal_compartments ~ "normal",
    cell_type_assignment_RNA_compartment == "fetal_nephron" & cell_type_assignment_RNA_celltype == "Cap mesenchyme" ~ "tumor",
    .default =  "unknown"
```

jashapiro commented on analyses/cell-type-wilms-tumor-14/scripts/summary_results.R:
```diff
@@ -89,7 +89,18 @@ arg_df <- tidyr::expand_grid(
 dfs <- purrr::pmap(arg_df,
              \(anchor_assay, level) concat_sample(anchor_assay, level, meta)
 )
+# add rough tumor cell classification, CM = tumor cells, immune/endothelium = normal
```
Adding the definition used below
```suggestion
# add rough tumor cell classification, CM = tumor cells, immune/endothelium = normal
normal_compartments <- c("immune", "endothelium")
```

jashapiro commented on analyses/cell-type-wilms-tumor-14/scripts/summary_results.R:
```diff
@@ -89,7 +89,18 @@ arg_df <- tidyr::expand_grid(
+    TRUE ~ "unknown"
+  ),
+  tumor_cell_classification_SCT = case_when(
+    cell_type_assignment_SCT_compartment == "immune" | cell_type_assignment_SCT_compartment == "endothelium" ~ "normal",
+    cell_type_assignment_SCT_compartment == "fetal_nephron" & cell_type_assignment_SCT_celltype == "Cap mesenchyme" ~ "tumor",
+    TRUE ~ "unknown"
```
```suggestion
    cell_type_assignment_SCT_compartment %in% normal_compartments ~ "normal",
    cell_type_assignment_SCT_compartment == "fetal_nephron" & cell_type_assignment_SCT_celltype == "Cap mesenchyme" ~ "tumor",
    .default = "unknown"
```

jashapiro reviewed (commented):
Thank you for adding this. 

I made some minor code suggestions, but can you also include some of your comments from the PR about how you are classifying tumor cells in the module readme?

Finally, while I appreciate the explicitness of including both the SCT and Lognormalized results for both compartment and cell type, I think it would be good to have a result column that gives "default" cell type calls. We had discussed some about using the compartment in some cases and filling in with cell types where appropriate, and I think that would be worth doing here, even if you are not fully satisfied with the result. You might also make a decision about whether to use the SCT or lognormal results by default? Or maybe only make calls when they agree? I did not get too much of a sense from the table alone how concordant they were.

JingxuanChen7 commented:
> I made some minor code suggestions, but can you also include some of your comments from the PR about how you are classifying tumor cells in the module readme?

Added to `results/README.md`.

> Finally, while I appreciate the explicitness of including both the SCT and Lognormalized results for both compartment and cell type, I think it would be good to have a result column that gives "default" cell type calls. We had discussed some about using the compartment in some cases and filling in with cell types where appropriate, and I think that would be worth doing here, even if you are not fully satisfied with the result. You might also make a decision about whether to use the SCT or lognormal results by default? Or maybe only make calls when they agree? I did not get too much of a sense from the table alone how concordant they were.

Thank you for putting forward our previous discussion on combining compartments! I have added another column `cell_type_assignment` to combine immune, endo, and stroma compartments. Regarding "default" column, I would use `logNormalize` results, since in general more anchors were identified in this set.

Also attempted to fix the CI issue. Hopefully it would work.

jashapiro commented on analyses/cell-type-wilms-tumor-14/scripts/summary_results.R:
```diff
@@ -89,7 +88,25 @@ arg_df <- tidyr::expand_grid(
+    cell_type_assignment_RNA_compartment %in% normal_compartments ~ "normal",
+    cell_type_assignment_RNA_compartment == "fetal_nephron" & cell_type_assignment_RNA_celltype == "Cap mesenchyme" ~ "tumor",
+    .default = "unknown"
+  ) ) %>%
+  { if ("SCT" %in% assays) mutate(., tumor_cell_classification_SCT = case_when(
+    cell_type_assignment_SCT_compartment %in% normal_compartments ~ "normal",
+    cell_type_assignment_SCT_compartment == "fetal_nephron" & cell_type_assignment_SCT_celltype == "Cap mesenchyme" ~ "tumor",
+    .default = "unknown"
+  ) ) else . } %>% 
+  # use logNormalize as "default" final result, combine immune, endo, and stroma compartments
+  mutate( cell_type_assignment = if_else(cell_type_assignment_RNA_compartment != "fetal_nephron", 
+                                         cell_type_assignment_RNA_compartment, 
+                                         cell_type_assignment_RNA_celltype ) ) %>%
+  mutate( tumor_cell_classification = tumor_cell_classification_RNA )
```
While what you have does work, I find that nesting in `if` statements is likely to lead to confusion, so I am breaking up the pipeline a bit for clarity. 

```suggestion
  ) )

if ("SCT" %in% assays) {
  final_table <- final_table %>%
    mutate( tumor_cell_classification_SCT = case_when(
      cell_type_assignment_SCT_compartment %in% normal_compartments ~ "normal",
      cell_type_assignment_SCT_compartment == "fetal_nephron" & cell_type_assignment_SCT_celltype == "Cap mesenchyme" ~ "tumor",
      .default = "unknown"
    ) )
}

final_table <- final_table %>%
  # use logNormalize as "default" final result, combine immune, endo, and stroma compartments
  mutate( cell_type_assignment = if_else(cell_type_assignment_RNA_compartment != "fetal_nephron", 
                                         cell_type_assignment_RNA_compartment, 
                                         cell_type_assignment_RNA_celltype ) ) %>%
  mutate( tumor_cell_classification = tumor_cell_classification_RNA )
```

jashapiro reviewed (approved):
Thank you for adding the readme comments. I have checked the summary table on S3, and everything seems to be as described.

Thanks again for yoru contribution, and adding this before the deadline!
